GliaWidgets/SecureConversations/SecureConversations.WelcomeView.swift

```diff
@@ -3,30 +3,483 @@ import UIKit
 
 extension SecureConversations {
     final class WelcomeView: View {
-        private let style: SecureConversations.WelcomeStyle
+        static let sideMargin = 24.0
+        static let filePickerButtonSize = 44.0
+
+        struct Props: Equatable {
+            let style: SecureConversations.WelcomeStyle
+            let backButtonTap: Cmd
+            let closeButtonTap: Cmd
+            let checkMessageButtonTap: Cmd
+            let filePickerButtonTap: Cmd?
+            let sendMessageButtonTap: Cmd
+            let messageTextViewProps: MessageTextView.Props
+        }
+
+        var props: Props {
+            didSet {
+               renderProps()
+            }
+        }
+
         let header: Header
+        let titleIconView = UIImageView().makeView { imageView in
+            imageView.image = Asset.mcEnvelope.image
+        }
+
+        let titleLabel = UILabel().makeView { label in
+            label.numberOfLines = 0
+        }
 
-        init(with style: SecureConversations.WelcomeStyle) {
-            self.style = style
-            self.header = Header(with: style.header)
+        let subtitleLabel = UILabel().makeView { label in
+            label.numberOfLines = 0
+        }
+
+        lazy var checkMessagesButton = UIButton(type: .custom).makeView { button in
+            button.addTarget(
+                self,
+                action: #selector(handleCheckMessagesButtonTap),
+                for: .touchUpInside
+            )
+        }
+
+        let messageTitleLabel = UILabel().makeView()
+
+        lazy var filePickerButton = UIButton().makeView { button in
+            button.setImage(Asset.mcPickMedia.image, for: .normal)
+            button.addTarget(
+                self,
+                action: #selector(handleFilePickerButtonTap),
+                for: .touchUpInside
+            )
+        }
 
+        let messageTextView = MessageTextView().makeView()
+
+        lazy var sendMessageButton = UIButton().makeView { button in
+            button.addTarget(
+                self,
+                action: #selector(handleSendMessageButtonTap),
+                for: .touchUpInside
+            )
+        }
+
+        lazy var scrollView = UIScrollView().makeView { scrollView in
+            scrollView.keyboardDismissMode = .onDrag
+            scrollView.addGestureRecognizer(scrollViewTapRecognizer)
+        }
+
+        lazy var rootStackView = UIStackView().makeView { stackView in
```

Comment:
```swift
lazy var rootStackView = UIStackView.make(.vertical, alignment: .leading)(/* here could be subviews */)
```

Reply:
@dukhovnyi  Yes, however not all properties (like `isLayoutMarginsRelativeArrangement` and `layoutMargins`) are present on this `make` method, so I still need to use configuration closure. However I think it is cool idea to have all arranged views added in one place at once, because it allows visually to understand the order of views in the stack. So added changes a bit differently 36182adf3af8ae77e1f600d48252be96e9e1a364.